Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -201,49 +201,59 @@ protected virtual Task ProcessNavigationForAbsoulteUri(Queue<string> segments, N
         {
             if (currentPage.Navigation.NavigationStack.Count == 0)
             {
-                var newRoot = CreatePageFromSegment(nextSegment);
-                await ProcessNavigation(newRoot, segments, parameters, false, animated);
-                await DoNavigateAction(currentPage, nextSegment, newRoot, parameters, async () =>
+                var nextPage = CreatePageFromSegment(nextSegment);
+                await ProcessNavigation(nextPage, segments, parameters, false, animated);
+                await DoNavigateAction(currentPage, nextSegment, nextPage, parameters, async () =>
                 {
-                    await DoPush(currentPage, newRoot, false, animated);
+                    await DoPush(currentPage, nextPage, false, animated);
                 });
                 return;
             }
+            if (!GetClearNavigationPageNavigationStack(currentPage))
+            {
+                var fromPage = currentPage.Navigation.NavigationStack.Last();
+                var nextPage = CreatePageFromSegment(nextSegment);
+                await ProcessNavigation(nextPage, segments, parameters, false, animated);
```

Comment:
Now when ClearNavigationStackOnNavigation is false and the current page is the same as the page being navigated to, you get a duplicate page.  You need to also check to make sure the target page is not the same type as the current page. Similar to line 223